lib/ansible/module_utils/basic.py

```diff
@@ -2329,6 +2329,8 @@ def _return_formatted(self, kwargs):
                         self.deprecate(d[0], version=d[1])
                     else:
                         self.deprecate(d)
+            elif isinstance(kwargs['deprecations'], dict):
+                self.deprecate(kwargs['deprecations']['message'], version=kwargs['deprecations']['version'])
```

Comment:
I have not looked at this in depth, but I believe this should be `msg` instead of `message`.

Reply:
Correct, I was fixing that before I saw your comment :-)